nixos/modules/services/games/pterodactyl-wings.nix

```diff
@@ -0,0 +1,214 @@
+              interface = mkOption {
+                type = str;
+                default = "172.18.0.1";
+                description = "The interface that should be used to create the network. Must not conflict with any other interfaces in use by Docker or on the system.";
+              };
+              dns = mkOption {
+                type = listOf str;
+                default = [ "1.1.1.1" "1.0.0.1" ];
+                description = "The DNS settings for containers.";
+              };
+            };
+          };
+          remote = mkOption {
+            type = str;
+            description = "The URL where the panel is running that this daemon should connect to, to collect data and send events.";
+          };
+          allowed_origins = mkOption {
+            type = listOf str;
+            default = [];
+            description = "AllowedOrigins is a list of allowed request origins. The Panel URL is automatically allowed, this is only needed for adding additional origins.";
+          };
+        };
+      };
+    };
+  };
+  config = lib.mkIf cfg.enable {
+    assertions = [ {
+      assertion = !cfg.settings.api.ssl.enabled || (cfg.settings.api.ssl.cert != null && cfg.settings.api.ssl.key != null);
+      message = "Certificate and key path must be set in config.services.pterodactyl-wings.settings.api.ssl to enable TLS on the API server";
+    } ];
+    warnings = lib.lists.optional (builtins.hasAttr "token" cfg.settings)
+    ''
+      A token has been defined at services.wings.settings.token.
+      This exposes it in the world-readable nix store, and will be
+      overridden by services.wings.tokenFile regardless.
+     '';
+
+    virtualisation.docker.enable = lib.mkDefault true;
+
+    users = lib.optionalAttrs (cfg.settings.system.username == "pterodactyl") {
+      users.pterodactyl = {
+        isSystemUser = true;
+        group = "pterodactyl";
+      };
+      groups.pterodactyl = {};
+    };
+
+    # https://pterodactyl.io/wings/1.0/installing.html#daemonizing-using-systemd
+    systemd.services.pterodactyl-wings = let
+      cfgFile = settingsFormat.generate "wings.yaml" cfg.settings;
+    in {
+      requires = [ "docker.service" ];
+      after = [ "docker.service" ];
+      partOf = [ "docker.service" ];
+      description = "Pterodactyl Wings Daemon";
+      serviceConfig = {
+        WorkingDirectory = "/etc/pterodactyl";
+        LimitNOFILE = 4096;
+        PIDFile = "/run/wings/daemon.pid";
+        RuntimeDirectory = "wings";
+        RuntimeDirectoryMode = "0700";
+        ExecStartPre = [
+          "${pkgs.coreutils}/bin/cp ${cfgFile} \${RUNTIME_DIRECTORY}/wings.yaml"
+          "${pkgs.bash}/bin/sh -c 'echo token: $(<${cfg.tokenFile}) >> \${RUNTIME_DIRECTORY}/wings.yaml'"
+        ];
+        ExecStart = "${pkgs.pterodactyl-wings}/bin/wings --config \${RUNTIME_DIRECTORY}/wings.yaml";
+        Restart = "on-failure";
+        RestartSec = "5s";
+      };
+      startLimitIntervalSec = 180;
```

Comment:
This seems quite excessive

Reply:
This was from their systemd service configuration, but I'll reduce both.